packages/grafana-ui/src/slate-plugins/suggestions.tsx

```diff
@@ -337,3 +346,35 @@ const handleTypeahead = async (
   // Bogus edit to force re-render
   editor.blur().focus();
 };
+
+export function getNumCharsToDelete(
+  suggestionText: string,
+  typeaheadPrefix: string,
+  typeaheadText: string,
+  preserveSuffix: boolean,
+  deleteBackwards?: number,
+  cleanText?: (text: string) => string
+) {
+  // remove the current, incomplete text and replace it with the selected suggestion
+  const backward = deleteBackwards || typeaheadPrefix.length;
+  const text = cleanText ? cleanText(typeaheadText) : typeaheadText;
+  const offset = typeaheadText.indexOf(typeaheadPrefix);
+
+  let forward: number;
+
+  if (window.grafanaBootData?.settings.featureToggles['slateAutocomplete']) {
+    const suffixLength =
+      offset > -1 ? text.length - offset - typeaheadPrefix.length : text.length - typeaheadPrefix.length;
+    const midWord = Boolean((typeaheadPrefix && suffixLength > 0) || suggestionText === typeaheadText);
+    forward = midWord && !preserveSuffix ? suffixLength + offset : 0;
+  } else {
+    const suffixLength = text.length - typeaheadPrefix.length;
+    const midWord = typeaheadPrefix && ((suffixLength > 0 && offset > -1) || suggestionText === typeaheadText);
+    forward = midWord && !preserveSuffix ? suffixLength + offset : 0;
```

Comment:
The `forward` logic is the same for both branches of if/else, so maybe it' be moved out of the condition?

Reply:
@Clarity-89 yeah i had that originally, but the main problem then is you have to define the `suffixLength` and `midWord` variables outside of the branches. `midWord` in particular is annoying because it has different types between the 2 branches... one is a pure `boolean`, whereas the other is something like `string | boolean` 😢

Reply:
yeah, good point.